src/rust/src/backend/ciphers.rs

```diff
@@ -113,9 +119,39 @@ impl CipherContext {
         Ok(CipherContext {
             ctx,
             py_mode: mode.into(),
+            side,
+            reset_nonce_allowed,
+            nonce_length: iv_nonce.map(|v| v.as_bytes().len()).unwrap_or(0) as u8,
         })
     }
 
+    fn reset_nonce(&mut self, nonce: CffiBuf<'_>) -> CryptographyResult<()> {
+        if !self.reset_nonce_allowed {
+            return Err(CryptographyError::from(
+                exceptions::UnsupportedAlgorithm::new_err((
+                    "This algorithm or mode does not support resetting the nonce.",
+                    exceptions::Reasons::UNSUPPORTED_CIPHER,
+                )),
+            ));
+        }
```

Comment:
Instead of storing this, please just do an `isinstance` check on `self.py_mode` here.